tests/unit_tests/cpu/utils/grids/axis.cpp

```diff
@@ -342,6 +342,64 @@ GTEST_TEST(detray_grid, closed_irregular_axis) {
     EXPECT_EQ(cir_axis.range(3.f, nhood11s), expected_range);
 }
 
+GTEST_TEST(detray_grid, axis_comparison) {
+
+    // Should be sufficient to test the comparison operators
+    using x_axis_op_r_t = single_axis<open<label::e_x>, regular<>>;
+    using x_axis_cl_r_t = single_axis<closed<label::e_x>, regular<>>;
+    using x_axis_ci_r_t = single_axis<circular<label::e_x>, regular<>>;
+    using x_axis_op_ir_t = single_axis<open<label::e_x>, irregular<>>;
+    using x_axis_cl_ir_t = single_axis<closed<label::e_x>, irregular<>>;
+    using x_axis_ci_orr_t = single_axis<circular<label::e_x>, irregular<>>;
+
+    std::tuple<x_axis_op_r_t, x_axis_cl_r_t, x_axis_ci_r_t, x_axis_op_ir_t,
+               x_axis_cl_ir_t, x_axis_ci_orr_t>
+        axes = {};
+
+    auto test_axis = [](auto axis) -> void {
+        using axis_type = decltype(axis);
+
+        // Lower bin edges: min and max bin edge for the regular axis
+        vecmem::vector<scalar> bin_edges = {-10.f, -5.f, -3.f, 7.f,
+                                            7.f,   14.f, 20.f};
+
+        vecmem::vector<scalar> different_bin_edges = {-10.f, -8.f, -3.f, 7.f,
+                                                      7.f,   14.f, 20.f};
+
+        // Regular axis: first entry is the offset in the bin edges vector (2),
+        // the second entry is the number of bins (10): Lower and upper bin
+        // edges of the max and min bin are -3 and 7 => stepsize is (7 - (-3)) /
+        // 10 = 1
+        // ... -3, -2, -1,  0,  1,  2,  3,  4,  5,  6,  7 ...
+        //   [0] [1] [2] [3] [4] [5] [6] [7] [8] [9] [10] [11]
+        const dindex_range edge_range = {2u, 10u};
+
+        const dindex_range different_edge_range = {2u, 8u};
+
+        // An open regular x-axis
+        axis_type or_axis{edge_range, &bin_edges};
+
+        axis_type another_or_axis{edge_range, &bin_edges};
+
+        axis_type differnt_or_axis{edge_range, &different_bin_edges};
```

Comment:
What happens if we have different edge ranges but on identical edge data. Meaning the same edge values in different spots on the bin_edges vector?